### ISSUE TITLE
Preserve focused workspace across monitor replug

### ISSUE THREAD

**Issue by kchibisov** [YaLTeR/niri#63]
After switching tty back and forth I always end up on the first workspace regardless of where I was before.

**Comment by kchibisov** [comment]
I've also noticed that focused client on HiDPI doesn't get a `enter` resend for surfaces, so the scale is `1` and won't fix itself until I switch TTY again.

**Comment by YaLTeR** [comment]
> I've also noticed that focused client on HiDPI doesn't get a enter resend for surfaces, so the scale is 1 and won't fix itself until I switch TTY again.

That's https://github.com/Smithay/smithay/issues/1138

**Comment by YaLTeR** [comment]
This has been fixed in a sense that after cbbb7a26fc0bff426d67bb2804167f01ade5fbe4, TTY switches no longer replug the monitors, so the focused workspace does not reset. However, it would be nice to also preserve the focused workspace across monitor replugs anyway, so I'm adjusting this issue.

> I've also noticed that focused client on HiDPI doesn't get a `enter` resend for surfaces, so the scale is 1 and won't fix itself until I switch TTY again.

Just for any future readers, I fixed this one in Smithay.
